modules/tinymce/src/core/main/ts/fmt/ListItemFormat.ts

```diff
@@ -32,25 +32,44 @@ const isRngEndAtEndOfElement = (rng: Range, elm: Element) => {
 
 const isEditableListItem = (dom: DOMUtils) => (elm: Element) => NodeType.isListItem(elm) && dom.isEditable(elm);
 
+const findParentListItem = (selection: EditorSelection, el: Element) => Optional.from(selection.dom.getParent(el, NodeType.isListItem));
+
 const getFullySelectedBlocks = (selection: EditorSelection) => {
   const blocks = selection.getSelectedBlocks();
   const rng = selection.getRng();
 
   if (selection.isCollapsed()) {
     return [];
-  } if (blocks.length === 1) {
+  }
+
+  // TINY-13197: The first block element returned by getSelectedBlocks() is not LI if the content is wrapped inside a block element, even when the content is fully selected. The following logic finds and replaces the first block with its parent LI
```

Comment:
Don't we have a line length lint restriction anymore? 😅

Reply:
I also suggest adding some clarity around this - this only needs to edit the first block because when there are multiple list items selected, the second and subsequent _do_ return the `li` as the selected block. So really this is just normalising the selection (and one way to implement this would've been dealing with this in the selection normalisation code instead of the list code).